workers/proto/kitchen_sink/kitchen_sink.proto

```diff
@@ -0,0 +1,328 @@
+    TimerAction timer = 1;
+    ExecuteActivityAction exec_activity = 2;
+    ExecuteChildWorkflowAction exec_child_workflow = 3;
+    AwaitSignalAction await_signal = 4;
+    AwaitUpdateAction await_update = 5;
+    SendSignalAction send_signal = 6;
+    SetPatchMarkerAction set_patch_marker = 7;
+    UpsertSearchAttributesAction upsert_search_attributes = 8;
+    UpsertMemoAction upsert_memo = 9;
+    WorkflowState set_workflow_state = 10;
+
+    ReturnResultAction return_result = 11;
+    ReturnErrorAction return_error = 12;
+    ContinueAsNewAction continue_as_new = 13;
+
+    ActionSet nested_action_set = 14;
+  }
+}
+
+// The following actions and the messages they depend on are largely duplicative of Core/Server's
+// commands, but some care is taken to remove things that would be uninteresting for these tests.
+
+// All cancellable commands will have this available as a field. If it is set, the command
+// should be cancelled at the specified juncture.
+message CancelCondition {
+  oneof condition {
+    google.protobuf.Empty before_started = 1;
+    google.protobuf.Empty after_started = 2;
+    google.protobuf.Empty after_completed = 3;
+  }
+}
+
+message TimerAction {
+  uint64 milliseconds = 1;
+}
+
+message ExecuteActivityAction {
+  string activity_id = 1;
+  string activity_type = 2;
+  // The name of the task queue to place this activity request in
+  string task_queue = 3;
+  map<string, temporal.api.common.v1.Payload> headers = 4;
+  // Arguments/input to the activity. Called "input" upstream.
+  repeated temporal.api.common.v1.Payload arguments = 5;
+  // Indicates how long the caller is willing to wait for an activity completion. Limits how long
+  // retries will be attempted. Either this or start_to_close_timeout_seconds must be specified.
+  // When not specified defaults to the workflow execution timeout.
+  google.protobuf.Duration schedule_to_close_timeout = 6;
+  // Limits time an activity task can stay in a task queue before a worker picks it up. This
+  // timeout is always non retryable as all a retry would achieve is to put it back into the same
+  // queue. Defaults to schedule_to_close_timeout or workflow execution timeout if not specified.
+  google.protobuf.Duration schedule_to_start_timeout = 7;
+  // Maximum time an activity is allowed to execute after a pick up by a worker. This timeout is
+  // always retryable. Either this or schedule_to_close_timeout must be specified.
+  google.protobuf.Duration start_to_close_timeout = 8;
+  // Maximum time allowed between successful worker heartbeats.
+  google.protobuf.Duration heartbeat_timeout = 9;
+  // Activities are provided by a default retry policy controlled through the service dynamic
+  // configuration. Retries are happening up to schedule_to_close_timeout. To disable retries set
+  // retry_policy.maximum_attempts to 1.
+  temporal.api.common.v1.RetryPolicy retry_policy = 10;
+  // Whether or not this activity will be a local activity
+  oneof locality {
```

Comment:
Given that normal activities have some options local activities do not does it make sense for them to be a separate action?

Reply:
We'd have to dupe everything else in the message which is why I went with that.

Reply:
Yeah I am arguing your not duplicating everything since not everything in a activity applies to a local activity. I think it would require less validation in the implementations to split them up, but not a big deal